### PR TITLE
Update README and CHANGELOG for 4.0.0 release

### PR DESCRIPTION
**Pull request by barchetta**
### Description

Updates CHANGELOG and README for Helidon 4

### Documentation

No impact

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,55 @@ For Helidon 2.x releases please see [Helidon 2.x CHANGELOG.md](https://github.co
 
 For Helidon 1.x releases please see [Helidon 1.x CHANGELOG.md](https://github.com/oracle/helidon/blob/helidon-1.x/CHANGELOG.md)
 
+## [4.0.0]
+
+We are pleased to announce the release of Helidon 4.0.0. The big news in Helidon 4.0.0 is the introduction of Helidon Níma -- a ground up webserver implementation based on JDK Project Loom virtual threads. With Helidon 4 you get the high throughput of a reactive server with the simplicity of thread-per-request style programming.
+
+The Helidon SE API in 4.0.0 has changed significantly from Helidon 3. The use of virtual threads have enabled these APIs to change from asynchronous to blocking. This results in much simpler code that is easier to write, maintain, debug and understand. Existing Helidon SE code will require modification to run on these new APIs. For more information see the [Helidon SE Upgrade Guide](https://helidon.io/docs/v4/#/se/guides/upgrade_4x).
```
Comment (ljnelson) [inline CHANGELOG.md:18]: Suggestion: make that first sentence bold

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,55 @@ For Helidon 2.x releases please see [Helidon 2.x CHANGELOG.md](https://github.co
...
+
+Helidon 4 supports MicroProfile 6. This means your existing Helidon MP 3.x applications will run on Helidon 4 with only minor modifications. And since Helidon’s MicroProfile server is based on the new Níma WebServer you get all the benefits of running on virtual threads. For more information see the [Helidon MP Upgrade Guide](https://helidon.io/docs/v4/#/mp/guides/upgrade_4x).
+
+New to Helidon? Then jump in and [get started](https://helidon.io/docs/v4/#/about/prerequisites).
+
+Java 21 is required to use Helidon 4.0.0
```
Comment (ljnelson) [inline CHANGELOG.md:24]: Missing trailing period?

File: README.md [hunk 3]
```diff
@@ -23,12 +23,13 @@ Helidon supports two programming models:
...
 In either case your application is a Java SE program running on the
 new Helidon Níma WebServer that has been written from the ground up to
-use Java 21 Virtual Threads.
+use Java 21 Virtual Threads. With Helidon 4 you get the high throughput of a reactive server with the simplicity of thread-per-request style programming.
 
-In Helidon 4 each request is processed by a dedicated virtual thread so
-your code is free to perform blocking operations without impacting your server's
-ability to handle other requests. You get all the throughput of a reactive
-server with none of the complexity.
+The Helidon SE API in Helidon 4 has changed significantly from Helidon 3. The use of virtual threads has enabled these APIs to change from asynchronous to blocking. This results in much simpler code that is easier to write, maintain, debug and understand. Earlier Helidon SE code will require modification to run on these new APIs. For more information see the [Helidon SE Upgrade Guide](https://helidon.io/docs/v4/#/se/guides/upgrade_4x).
```
Comment (ljnelson) [inline README.md:28]: Make first sentence bold?

File: CHANGELOG.md (hunk 1 above)
Comment (barchetta) [reply CHANGELOG.md:18]: I understand your point, but my preference is for less busy typography. So I'm going to leave out the bold.

File: CHANGELOG.md (hunk 2 above)
Comment (barchetta) [reply CHANGELOG.md]: Fixed.

**Review by ljnelson — APPROVED** [review]

**Pull request merged**